src/GafferScene/ShaderTweaks.cpp

```diff
@@ -316,8 +323,66 @@ bool ShaderTweaks::applyTweaks( IECoreScene::ShaderNetwork *shaderNetwork, Tweak
 				{
 					throw IECore::Exception( fmt::format( "Cannot apply tweak to \"{}\" : Mode must be \"Replace\" when inserting a connection", name ) );
 				}
+
 				const auto inputParameter = ShaderNetworkAlgo::addShaders( shaderNetwork, inputNetwork );
 				shaderNetwork->addConnection( { inputParameter, parameter } );
+
+				// TODO - it would be substantially more efficient to search for and process tweak sources
+				// just in `inputNetwork` before merging it to `shaderNetwork` ... but this would require
+				// dealing with weird connections where the input node handle is relative to `shaderNetwork`,
+				// but the output handle is relative to `inputNetwork`. This gets very confusing if there are
+				// nodes in the two networks with the same name, which get uniquified during addShaders.
+				// Doing this after merging simplifies all that.
+				std::vector<IECore::InternedString> shadersToDelete;
+				for( const auto &i : shaderNetwork->shaders() )
+				{
+					if( i.second->getName() == ShaderTweakProxy::shaderTweakProxyIdentifier() )
+					{
+						shadersToDelete.push_back( i.first );
```

Comment:
Seems like we never actually delete these.

Reply:
Oh, and maybe we don't need to, because we remove the output connections from the proxy, and therefore hit `removeUnusedShaders()` at the end of this function?

Reply:
Yeah, I was assuming I didn't need to worry about it if we're doing a removeUnused anyway.

Reply:
Looks like we are still filling an unnecessary `shadersToDelete` container that we never actually use.